tests/mocha/procedure_map_test.js

```diff
@@ -200,8 +201,494 @@ suite('Procedure Map', function() {
     });
   });
 
-  suite('event firing', function() {
-    // TBA after the procedure map is implemented
+  suite.skip('event firing', function() {
```

Comment:
Is this supposed to be skipped?

Reply:
Yesh! This adds tests for firing events, the next PR will actually add the firing of said events.